libsodium-sys/build.rs

```diff
@@ -146,139 +134,46 @@ fn find_libsodium_pkg() {
-    let mut zip_archive = ZipArchive::new(compressed_file).unwrap();
-
-    // Extract just the appropriate version of libsodium.lib and headers to the install path.  For
-    // now, only handle MSVC 2015.
-    let arch_path = if cfg!(target_pointer_width = "32") {
-        Path::new("Win32")
-    } else if cfg!(target_pointer_width = "64") {
-        Path::new("x64")
-    } else {
-        panic!("target_pointer_width not 32 or 64")
-    };
-
-    let unpacked_lib = arch_path.join("Release/v140/static/libsodium.lib");
-    for i in 0..zip_archive.len() {
-        let mut entry = zip_archive.by_index(i).unwrap();
-        let entry_name = entry.name().to_string();
-        let entry_path = Path::new(&entry_name);
-        let opt_install_path = if entry_path.starts_with("include") {
-            let is_dir = (entry.unix_mode().unwrap() & S_IFDIR as u32) != 0;
-            if is_dir {
-                let _ = fs::create_dir(&Path::new(&install_dir).join(entry_path));
-                None
-            } else {
-                Some(Path::new(&install_dir).join(entry_path))
-            }
-        } else if entry_path == unpacked_lib {
-            Some(lib_install_dir.join("libsodium.lib"))
-        } else {
-            None
-        };
-        if let Some(full_install_path) = opt_install_path {
-            let mut buffer = Vec::with_capacity(entry.size() as usize);
-            assert_eq!(entry.size(), entry.read_to_end(&mut buffer).unwrap() as u64);
-            let mut file = File::create(&full_install_path).unwrap();
-            file.write_all(&buffer).unwrap();
-        }
-    }
-
     println!("cargo:rustc-link-lib=static=libsodium");
-    println!(
-        "cargo:rustc-link-search=native={}",
-        lib_install_dir.display()
-    );
+    println!("cargo:rustc-link-search=native=msvc/Win32/Release/v140/");
 }
 
-#[cfg(all(windows, not(target_env = "msvc")))]
+#[cfg(all(target_env = "msvc", target_pointer_width = "64"))]
 fn build_libsodium() {
-    use libflate::gzip::Decoder;
-    use tar::Archive;
-
-    // Download gz tarball
-    let install_dir = get_install_dir();
-    let lib_install_dir = Path::new(&install_dir).join("lib");
-    fs::create_dir_all(&lib_install_dir).unwrap();
-    let url = format!("{}libsodium-{}-mingw.tar.gz", DOWNLOAD_BASE_URL, VERSION);
-    let compressed_file = download(&url, SHA256);
-
-    // Unpack the tarball
-    let gz_decoder = Decoder::new(compressed_file).unwrap();
-    let mut archive = Archive::new(gz_decoder);
-
-    // Extract just the appropriate version of libsodium.a and headers to the install path
-    let arch_path = if cfg!(target_pointer_width = "32") {
-        Path::new("libsodium-win32")
-    } else if cfg!(target_pointer_width = "64") {
-        Path::new("libsodium-win64")
-    } else {
-        panic!("target_pointer_width not 32 or 64")
-    };
+    println!("cargo:rustc-link-lib=static=libsodium");
+    println!("cargo:rustc-link-search=native=msvc/x64/Release/v140/");
```

Comment:
We link `Release` lib always but we can link `Debug` in debug mode. I could not find how to detect build mode in build.rs.